libtiledbsoma/src/soma/soma_array.h

```diff
@@ -733,6 +734,41 @@ class SOMAArray : public SOMAObject {
         }
     }
 
+    /**
+     * Retrieves the non-empty domain from the array. This is the union of the
+     * non-empty domains of the array fragments. Return std::nullopt for empty
+     * domains.
+     */
+    template <typename T>
+    std::optional<std::pair<T, T>> non_empty_domain_slot_opt(
```

Comment:
This is fine (and thanks for working on this!!) -- but -- I'd love for us to make `non_empty_domain` return `None` when no data have been written. I can do that as a follow-on on #2407 -- or you can ...

Reply:
Yes that would be a good follow-up. I was running into a lot of trouble with converting between the Arrow C structs and pybind11 with `std::optional` / `py::none` so opened this for now to make sure it's working for spatial.

Reply:
@nguyenv that is awesome! :)